lib/puppet/functions/reboot/wait.rb

```diff
@@ -5,20 +5,24 @@
 #
 # This has no valid use outside plans!
 Puppet::Functions.create_function(:'reboot::wait') do
+  local_types do
+    type 'TargetOrTargets = Variant[String[1], Target, Array[TargetOrTargets]]'
```

Comment:
Is there a way for this to refer directly to Boltlib::Targetspec (I don't think there is but maybe there's magic I don't know about). If not should we discuss adding this as a type to puppet?

Reply:
I don't know of a way myself, or at least not a way that I also know would pass rspec-puppet tests. If anybody else knows a way though I'd be happy to learn about it!

Reply:
TargetSpec will work in the plan but you'll have to make sure it's defined for tests.